linker/Mono.Linker/LinkContext.cs

```diff
@@ -243,9 +243,9 @@ static bool IsCore (AssemblyNameReference name)
 
 		public AssemblyDefinition [] GetAssemblies ()
 		{
-			IDictionary cache = _resolver.AssemblyCache;
```

Comment:
I think this change is wrong GetAssemblies is mostly used to find types which this change breaks for referenced assemblies. You should introduce another method instead